src/redux/slices.ts

```diff
@@ -1,12 +1,27 @@
 import { createSlice } from '@reduxjs/toolkit';
+import { Record } from '../types/records';
 
-const counterSlice = createSlice({
-    name: 'counter',
-    initialState: 0,
+type RecordsState = {
+    records: Record[];
+    isLoading: boolean;
+};
+
+let initialState: RecordsState = {
+    records: [],
+    isLoading: true,
+};
+
+const recordsSlice = createSlice({
+    name: 'recordsState',
+    initialState,
     reducers: {
-        increment: state => state + 1,
-        decrement: state => state - 1,
+        fetched: (state, action) => {
+            return {
+                records: action.payload.data,
+                isLoading: action.payload.loading,
```

Comment:
The isLoading value does not need to come from the payload. It can be assumed by the fact that we are updating the data. Alos - with redux-toolkit you don't need to create a new state object. You can mutate the state and behind the scenes it will do the immutable stuff

Reply:
Force of habit :)